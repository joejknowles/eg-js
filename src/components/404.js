import React from 'react';

export default () => (
    <h1>404 - page not found. Oops</h1>
);
