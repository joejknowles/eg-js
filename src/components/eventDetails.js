import React, { Component } from 'react';
import { connect } from 'react-redux';

import { createEventSelector } from '../reducers';
import { fetchEventDetailsRequest } from '../actions';

export class EventDetails extends Component {
  componentDidMount() {
    this.props.fetchEventDetails();
  }

  render() {
    const props = this.props;
    return (
     <div key={ props.id } className="event-details">
       <img src={ props.image } className="event-image" alt="event" />
       <h3 className="event-title">{ props.title }</h3>
       <p>
         { props.date }, at { props.venue }
       </p>
     </div>
   );
  }
}

const mapStateToProps = (state, { params: { id } }) => (
  createEventSelector(id)
);

const mapStateToDispatch = (dispatch, { params: { id } }) => ({
  fetchEventDetails: () => dispatch(fetchEventDetailsRequest(id))
});

export default connect(mapStateToProps, mapStateToDispatch)(EventDetails);
