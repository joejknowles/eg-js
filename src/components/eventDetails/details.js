import React from 'react';
import ExtraDetails from './extraDetails';
import LoadingIndicator from '../loadingIndicator';

export default (props) => {
  const anyLoaded = props.hasOwnProperty('venue');
  if (!anyLoaded) return <LoadingIndicator />;
  const extraDetailsLoaded = props.hasOwnProperty('unixdate');
  return (
    <span>
      <ul className="basic-info">
        <li>{ props.date }</li>
        <li>{ props.venue }</li>
      </ul>
      { extraDetailsLoaded ? (<ExtraDetails { ...props } />) : (<LoadingIndicator />) }
    </span>
  );
}
