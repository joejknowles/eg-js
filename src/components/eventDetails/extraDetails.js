import React from 'react';
import format from 'date-fns/format';

export default ({
    url, type, unixdate, information_title, information_description
}) => (
  <span>
    <ul>
      <li>{ format(unixdate, 'hh:mm A') }</li>
      <li><a href={ url }>View on Ticket Arena</a></li>
      <li>type: { type }</li>
    </ul>
    <h4>
      { information_title }
    </h4>
    <div dangerouslySetInnerHTML={ { __html: information_description } } />
  </span>
);
