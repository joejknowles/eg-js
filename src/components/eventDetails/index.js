import React, { Component } from 'react';
import { connect } from 'react-redux';

import Details from './details';

import { createEventSelector } from '../../reducers';
import { fetchEventDetailsRequest } from '../../actions';

export class EventDetails extends Component {
  componentDidMount() {
    this.props.fetchEventDetails();
  }

  render() {
    const props = this.props;
    return (
      <div key={ props.id } className="event-details">
        <img src={ props.image } className="event-image" alt="event" />
        <div className="event-info">
          <h2 className="event-title" dangerouslySetInnerHTML={ { __html: props.title } }/>
          <Details { ...props } />
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, { params: { id } }) => (
  createEventSelector(id)
);

const mapStateToDispatch = (dispatch, { params: { id } }) => ({
  fetchEventDetails: () => dispatch(fetchEventDetailsRequest(id))
});

export default connect(mapStateToProps, mapStateToDispatch)(EventDetails);
