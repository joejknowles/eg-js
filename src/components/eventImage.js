import React from 'react';

const defaultImageUrl = 'https://placehold.it/100x100?text=IMAGE';

export default ({ url }) => (
  <img
    src={ url || defaultImageUrl }
    className="event-image"
    alt="event"
  />
);
