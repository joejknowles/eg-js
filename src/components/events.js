import React, { Component } from 'react';
import { connect } from 'react-redux';

import Listing from './listing';
import LoadingIndicator from './loadingIndicator';
import { eventsSelector, eventsLoadingSelector } from '../reducers'

export class Events extends Component {
  componentDidMount() {
    this.props.fetchEvents();
  }

  render() {
    const { props } = this;
    if (props.eventsLoading) return (<LoadingIndicator />);
    return (
      <div className="events-list">
        {
          props.events.map(
            id => <Listing id={ id } key={ id } />
          )
        }
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  events: eventsSelector(state),
  eventsLoading: eventsLoadingSelector(state)
});

const mapDispatchToProps = (dispatch) => ({
  fetchEvents: () => dispatch({ type: "FETCH_EVENTS_REQUEST" })
});

export default connect(mapStateToProps, mapDispatchToProps)(Events);
