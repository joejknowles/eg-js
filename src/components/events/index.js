import React, { Component } from 'react';
import { connect } from 'react-redux';

import Listing from '../listing';
import LoadingIndicator from '../loadingIndicator';
import { eventsFirstLoadingSelector } from '../../reducers';
import { filteredEventsSelector } from '../../selectors/filter'

export class Events extends Component {
  componentDidMount() {
    this.props.fetchEvents();
  }

  render() {
    const { props } = this;
    if (props.eventsLoading) return (<LoadingIndicator />);
    return (
      <div className="events-list">
        {
          props.events.map(
            id => <Listing id={ id } key={ id } />
          )
        }
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  events: filteredEventsSelector(state),
  eventsLoading: eventsFirstLoadingSelector(state)
});

const mapDispatchToProps = (dispatch) => ({
  fetchEvents: () => dispatch({ type: "FETCH_EVENTS_REQUEST" })
});

export default connect(mapStateToProps, mapDispatchToProps)(Events);
