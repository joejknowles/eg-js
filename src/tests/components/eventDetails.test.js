import { EventDetails } from '../../components/eventDetails';

import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import { shallow } from 'enzyme';

const shallowRenderer = ReactTestUtils.createRenderer();

describe('EventDetails component', () => {
  const detailProps = {
    "id":"266970","title":"The Warehouse Project || Tuskegee","date":"Tuesday 27th December","image":"http://d13rtmsmlweq99.cloudfront.net/cms_media/images/events/948e6b789115dac236b33b5f606600ef.1467646509.jpg","type":"club","venue":"Store Steet, Manchester"
  };

  it('renders as before', () => {
    const component = shallowRenderer.render(
      <EventDetails { ...detailProps } />
    );
    expect(component).toMatchSnapshot();
  });

  it('calls fetchEventDetails on componentDidMount', () => {
    const fetchEventDetailsMock = jest.fn();
    const component = shallow(
      <EventDetails
        events={ [] }
        fetchEventDetails={ fetchEventDetailsMock }
      />
    );
    component.instance().componentDidMount();
    expect(fetchEventDetailsMock).toHaveBeenCalledWith("266970");
  });
});
