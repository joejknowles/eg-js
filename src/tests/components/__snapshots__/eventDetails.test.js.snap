exports[`EventDetails component renders as before 1`] = `
<div
  className="event-details">
  <img
    alt="event"
    className="event-image"
    src="http://d13rtmsmlweq99.cloudfront.net/cms_media/images/events/948e6b789115dac236b33b5f606600ef.1467646509.jpg" />
  <h3
    className="event-title">
    The Warehouse Project || Tuskegee
  </h3>
  <p>
    Tuesday 27th December
    , at 
    Store Steet, Manchester
  </p>
</div>
`;
